bootstrap/zerotrust/zerotrust.go

```diff
@@ -74,6 +75,18 @@ func HttpTransportFromClient(secretProvider interfaces.SecretProviderExt, client
 	return roundTripper, nil
 }
 
+type ZitiDialer struct {
+	underlayDialer *net.Dialer
+}
+
+func (z ZitiDialer) Dial(network, address string) (net.Conn, error) {
+	dialer := net.Dialer{
```

Comment:
i didn't want to invest too much time here yet, making this more configurable but here is one place that you might be able to inject another dialer. Then elsewhere you can set the fallback dialer.  Possibly "good enough" for now, but i'm open to other ideas too that are easier/more elegant. i just wanted to get the "big idea" out here for dicussion